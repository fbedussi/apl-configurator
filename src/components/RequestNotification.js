import React from 'react';

import LinkHome from './LinkHome';

const RequestNotification = ({labels, requestStatus}) => {
    if (requestStatus === 'unsent') {
        return null;
    }

    return (
        <div className="wrapper requestNotification">
            <div className={`requestNotification-sentWrapper ${requestStatus === 'sent' ? 'show' : 'hide'}`}>
                <p className="requestNotification-success">
                    {labels.requestSent}
                </p>
            </div>
            <div className={`requestNotification-successWrapper ${requestStatus === 'success' ? 'show' : 'hide'}`}>
                <p className="requestNotification-thanks">
                    {labels.thanks}
                </p>
                <p className="requestNotification-success">
                    {labels.requestSuccess}
                </p>
                <p className="requestNotification-followUp">
                    {labels.followUp}
                </p>
            </div>
            <div className={`requestNotification-errorWrapper ${requestStatus === 'error' ? 'show' : 'hide'}`}>
                <p className="requestNotification-error" dangerouslySetInnerHTML={{__html: labels.requestError}}>
                </p>
            </div>
            <LinkHome text={labels.linkHome} />
        </div>
    );
};

export default RequestNotification;
