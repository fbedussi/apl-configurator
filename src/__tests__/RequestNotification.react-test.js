import React from 'react';
import RequestNotification from '../components/RequestNotification';
import renderer from 'react-test-renderer';

const labels = {
        "back": "Indietro",
        "configurator": "Configuratore",
        "logoAlt": "torna al sito attraversamentipedonali.it",
        "logoDPowerAlt": "Attraversamento Pedonale Luminoso è un progetto D-power",
        "mandatory": "* campi obbligatori",
        "name": "Nome",
        "surname": "Cognome",
        "company": "Società (ragione sociale)",
        "address": "Indirizzo",
        "phone": "Telefono",
        "email": "E-mail",
        "website": "Sito",
        "reason": "Motivo della richiesta",
        "project": "Elaborazione progetto",
        "call": "Miglioria per gara d'appalto",
        "other": "Altro",
        "msg": "Eventuali informazioni aggiuntive",
        "submit": "Invia",
        "quotationBtn": "scopri quanto costa",
        "question": "domanda",
        "answer": "risposta",
        "thanks": "Grazie!",
        "requestSent": "Stiamo inviando la tua richiesta... attendi un attimo",
        "requestSuccess": "La vostra richiesta è stata inviata con successo",
        "requestError": "Purtroppo non siamo riusciti ad inviare la vostra richiesta, riprovate tra qualche minuto. Se il problema persiste vi preghiamo di segnalarcelo scrivendo a <a mailto=\"info@d-power.com\">info@d-power.com</a>.",
        "followUp": "Vi contatteremo al più presto",
        "linkHome": "Home page >",
        "homeUrl": "http://www.attraversamentipedonali.it/"
    };

test('RequestNotification component - unsent', () => {
  const requestStatus = 'unsnet';
  
  const component = renderer.create(
    <RequestNotification labels={labels} requestStatus={requestStatus} />
  );
  
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('RequestNotification component - sent', () => {
  const requestStatus = 'sent';

  const component = renderer.create(
    <RequestNotification labels={labels} requestStatus={requestStatus} />
  );
  
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('RequestNotification component - success', () => {
  const requestStatus = 'success';
  
  const component = renderer.create(
    <RequestNotification labels={labels} requestStatus={requestStatus} />
  );
  
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('RequestNotification component - error', () => {
  const requestStatus = 'error';
  
  const component = renderer.create(
    <RequestNotification labels={labels} requestStatus={requestStatus} />
  );
  
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});