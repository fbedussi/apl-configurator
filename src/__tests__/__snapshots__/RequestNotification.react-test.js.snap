// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestNotification component - error 1`] = `
<div
  className="wrapper requestNotification"
>
  <div
    className="requestNotification-sentWrapper hide"
  >
    <p
      className="requestNotification-success"
    >
      Stiamo inviando la tua richiesta... attendi un attimo
    </p>
  </div>
  <div
    className="requestNotification-successWrapper hide"
  >
    <p
      className="requestNotification-thanks"
    >
      Grazie!
    </p>
    <p
      className="requestNotification-success"
    >
      La vostra richiesta è stata inviata con successo
    </p>
    <p
      className="requestNotification-followUp"
    >
      Vi contatteremo al più presto
    </p>
  </div>
  <div
    className="requestNotification-errorWrapper show"
  >
    <p
      className="requestNotification-error"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Purtroppo non siamo riusciti ad inviare la vostra richiesta, riprovate tra qualche minuto. Se il problema persiste vi preghiamo di segnalarcelo scrivendo a <a mailto=\\"info@d-power.com\\">info@d-power.com</a>.",
        }
      }
    />
  </div>
  <a
    className="linkHome"
    href={undefined}
  >
    Home page &gt;
  </a>
</div>
`;

exports[`RequestNotification component - sent 1`] = `
<div
  className="wrapper requestNotification"
>
  <div
    className="requestNotification-sentWrapper show"
  >
    <p
      className="requestNotification-success"
    >
      Stiamo inviando la tua richiesta... attendi un attimo
    </p>
  </div>
  <div
    className="requestNotification-successWrapper hide"
  >
    <p
      className="requestNotification-thanks"
    >
      Grazie!
    </p>
    <p
      className="requestNotification-success"
    >
      La vostra richiesta è stata inviata con successo
    </p>
    <p
      className="requestNotification-followUp"
    >
      Vi contatteremo al più presto
    </p>
  </div>
  <div
    className="requestNotification-errorWrapper hide"
  >
    <p
      className="requestNotification-error"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Purtroppo non siamo riusciti ad inviare la vostra richiesta, riprovate tra qualche minuto. Se il problema persiste vi preghiamo di segnalarcelo scrivendo a <a mailto=\\"info@d-power.com\\">info@d-power.com</a>.",
        }
      }
    />
  </div>
  <a
    className="linkHome"
    href={undefined}
  >
    Home page &gt;
  </a>
</div>
`;

exports[`RequestNotification component - success 1`] = `
<div
  className="wrapper requestNotification"
>
  <div
    className="requestNotification-sentWrapper hide"
  >
    <p
      className="requestNotification-success"
    >
      Stiamo inviando la tua richiesta... attendi un attimo
    </p>
  </div>
  <div
    className="requestNotification-successWrapper show"
  >
    <p
      className="requestNotification-thanks"
    >
      Grazie!
    </p>
    <p
      className="requestNotification-success"
    >
      La vostra richiesta è stata inviata con successo
    </p>
    <p
      className="requestNotification-followUp"
    >
      Vi contatteremo al più presto
    </p>
  </div>
  <div
    className="requestNotification-errorWrapper hide"
  >
    <p
      className="requestNotification-error"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Purtroppo non siamo riusciti ad inviare la vostra richiesta, riprovate tra qualche minuto. Se il problema persiste vi preghiamo di segnalarcelo scrivendo a <a mailto=\\"info@d-power.com\\">info@d-power.com</a>.",
        }
      }
    />
  </div>
  <a
    className="linkHome"
    href={undefined}
  >
    Home page &gt;
  </a>
</div>
`;

exports[`RequestNotification component - unsent 1`] = `
<div
  className="wrapper requestNotification"
>
  <div
    className="requestNotification-sentWrapper hide"
  >
    <p
      className="requestNotification-success"
    >
      Stiamo inviando la tua richiesta... attendi un attimo
    </p>
  </div>
  <div
    className="requestNotification-successWrapper hide"
  >
    <p
      className="requestNotification-thanks"
    >
      Grazie!
    </p>
    <p
      className="requestNotification-success"
    >
      La vostra richiesta è stata inviata con successo
    </p>
    <p
      className="requestNotification-followUp"
    >
      Vi contatteremo al più presto
    </p>
  </div>
  <div
    className="requestNotification-errorWrapper hide"
  >
    <p
      className="requestNotification-error"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Purtroppo non siamo riusciti ad inviare la vostra richiesta, riprovate tra qualche minuto. Se il problema persiste vi preghiamo di segnalarcelo scrivendo a <a mailto=\\"info@d-power.com\\">info@d-power.com</a>.",
        }
      }
    />
  </div>
  <a
    className="linkHome"
    href={undefined}
  >
    Home page &gt;
  </a>
</div>
`;
